tests/core/test_bounding_rectangle.py

```diff
@@ -0,0 +1,124 @@
+        br4 = BoundingRectangle(-1.0, 6.0, -1.0, 6.0)
+        self.assertEqual(self.br.intersection_area(br4), 35.0)
+
+    def test_distance_to_b_rectangle(self):
+        br2 = BoundingRectangle(1.0, 4.0, 1.0, 4.0)
+        self.assertEqual(self.br.distance_to_b_rectangle(br2), 0.0)
+        br3 = BoundingRectangle(-4.0, -1.0, -5.0, -2.0)
+        self.assertEqual(self.br.distance_to_b_rectangle(br3), 1.0)
+        br4 = BoundingRectangle(7.0, 10.0, 9.0, 10.0)
+        self.assertEqual(self.br.distance_to_b_rectangle(br4), 5**0.5)
+        br5 = BoundingRectangle(-4.0, -1.0, 9.0, 10.0)
+        self.assertEqual(self.br.distance_to_b_rectangle(br5), 2**0.5)
+
+    def test_distance_to_point(self):
+        p0 = volmdlr.O2D
+        self.assertEqual(self.br.distance_to_point(p0), 0.0)
+        p1 = volmdlr.Point2D(1.0, 1.0)
+        self.assertEqual(self.br.distance_to_point(p1), 1.0)
+        p3 = volmdlr.Point2D(-3.0, 0.0)
+        self.assertEqual(self.br.distance_to_point(p3), 3.0)
+        p4 = volmdlr.Point2D(2.0, -5.0)
+        self.assertEqual(self.br.distance_to_point(p4), 3.0)
+        p5 = volmdlr.Point2D(6.0, 9.0)
+        self.assertEqual(self.br.distance_to_point(p5), 2**0.5)
+
+
+class TestBoundingRectangle2(unittest.TestCase):
+    b_rectangle1 = BoundingRectangle(-0.5, 1.0, -0.5, 1)
+    b_rectangle2 = BoundingRectangle(-1.0, 1.0, -1.0, 1.0)
+    b_rectangle3 = BoundingRectangle(4.0, 6.0, -1.0, 1.0)
+
+    def test_area(self):
+        self.assertEqual(self.b_rectangle1.area(), 2.25)
+
+    def test_center(self):
+        self.assertEqual(self.b_rectangle1.center(), volmdlr.Point2D(0.25, 0.25))
+
+    def test_intersection(self):
+        self.assertTrue(self.b_rectangle1.b_rectangle_intersection(self.b_rectangle2))
+        self.assertFalse(self.b_rectangle1.b_rectangle_intersection(self.b_rectangle3))
+
+    def test_is_inside(self):
+        self.assertTrue(self.b_rectangle1.is_inside_b_rectangle(self.b_rectangle2))
+        self.assertFalse(self.b_rectangle1.is_inside_b_rectangle(self.b_rectangle3))
+
+    def test_point_belongs(self):
+        self.assertTrue(self.b_rectangle1.point_belongs(volmdlr.Point2D(0.25, 0.25)))
+        self.assertFalse(self.b_rectangle1.point_belongs(volmdlr.Point2D(1, 1)))
+
+    def test_intersection_area(self):
+        self.assertEqual(self.b_rectangle1.intersection_area(self.b_rectangle2), 2.25)
+        self.assertEqual(self.b_rectangle1.intersection_area(self.b_rectangle3), 0)
+
+    def test_distance_to_point(self):
+        self.assertEqual(self.b_rectangle1.distance_to_point(volmdlr.Point2D(0.25, 0.25)), 0.75)
+        self.assertEqual(self.b_rectangle1.distance_to_point(volmdlr.Point2D(0.25, 2)), 1)
+
+    def test_distance_to_b_rectangle(self):
+        self.assertEqual(self.b_rectangle1.distance_to_b_rectangle(self.b_rectangle2), 0)
+        self.assertEqual(self.b_rectangle2.distance_to_b_rectangle(self.b_rectangle3), 3)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Why are there two classes testing it?

Reply:
There are two classes there because I realized after writing these tests that an other file (in an other directory) was already testing the class `BoundingRectangle`. So I compiled all the tests in a single file, but I can also combine the tests in a single class.